packages/workflow/src/interfaces.ts

```diff
@@ -1,39 +1,140 @@
-import { checkExtends, CommonWorkflowOptions } from '@temporalio/internal-workflow-common';
+import { RetryPolicy, TemporalFailure } from '@temporalio/common';
+import { checkExtends, CommonWorkflowOptions, SearchAttributeValue } from '@temporalio/internal-workflow-common';
 import type { coresdk } from '@temporalio/proto';
 
 /**
- * Workflow execution information
+ * Workflow Execution information
  */
 export interface WorkflowInfo {
   /**
    * ID of the Workflow, this can be set by the client during Workflow creation.
    * A single Workflow may run multiple times e.g. when scheduled with cron.
    */
   workflowId: string;
+
   /**
    * ID of a single Workflow run
    */
   runId: string;
 
   /**
-   * Filename containing the Workflow code
+   * Workflow function's name
    */
-  workflowType: string;
+  type: string;
 
   /**
-   * Namespace this Workflow is scheduled in
+   * Indexed information attached to the Workflow Execution
    */
-  namespace: string;
+  searchAttributes?: Record<string, SearchAttributeValue[]>;
 
   /**
-   * Task queue this Workflow is scheduled in
+   * Non-indexed information attached to the Workflow Execution
    */
-  taskQueue: string;
+  memo?: Record<string, unknown>;
+
+  /**
+   * Parent Workflow info (present if this is a Child Workflow)
+   */
+  parent?: ParentWorkflowInfo;
+
+  /**
+   * Result from the previous Run (present if this is a Cron Workflow or was Continued As New).
+   *
+   * An array of values, since other SDKs may return multiple values from a Workflow.
+   */
+  lastResult?: unknown[];
+
+  /**
+   * Failure from the previous Run (present when this Run is a retry, or the last Run of a Cron Workflow failed)
+   */
+  lastFailure?: TemporalFailure;
 
   /**
-   * Whether a Workflow is replaying history or processing new events
+   * Task queue this Workflow is executing on
    */
-  isReplaying: boolean;
+  taskQueue: string;
+
+  /**
+   * Less commonly used information
+   */
+  more: {
```

Comment:
Personally I think these can just move to the top level. There's no need to hide this info IMO. I can't think of a good reason to hide this stuff (a vertical scrollbar in intellisense is ok). Having said that, we shouldn't just add everything in workflow info, only valid stuff that represents information about the workflow run (_not_ the workflow task).

Reply:
> not the workflow task

Looks like `taskTimeout` is the only task-specific info? I like the idea of a Workflow being able to know what settings it was started with, whether it's just used for logging or impacts logic.